include/libtorrent/hasher512.hpp

```diff
@@ -92,6 +92,7 @@ namespace libtorrent {
 		explicit hasher512(span<char const> data);
 		hasher512(hasher512 const&);
 		hasher512& operator=(hasher512 const&) &;
+		hasher512& operator=(span<char const> data) = delete;
```

Comment:
I should take a look. technically this is a redundant change, I made it because:

1. cppcheck likes the symmetry of constructors and assignment operators
2. currently the only thing preventing assigning an arbitrary range is that the constructor is `explicit`, this is a more explicit way to say that's not allowed

So, that's to say I don't feel very strongly about this one.